web/controllers/screeningController/index.ts

```diff
@@ -347,6 +374,53 @@ export async function updateCourse(req: Request, res: Response) {
     }
 }
 
+async function handleNewLectures(lectures: { subcourse: { id: number }, start: Date, duration: number, instructor: { id: number } }[], courseId: number) {
+    const entityManager = getManager();
+
+    for (let lecture of lectures){
+        const course = await entityManager.findOne(Course, { id: courseId });
+        if (course == undefined) {
+            logger.warn(`No course found with ID ${courseId}`);
+            return 404;
+        }
+        const subcourse = await entityManager.findOne(Subcourse, { id: lecture.subcourse.id, course: course });
+        if (subcourse == undefined) {
+            logger.warn(`No subcourse found with ID ${lecture.subcourse.id}`);
+            return 404;
+        }
+
+        try {
+            await subcourse.addLecture(lecture);
+            await entityManager.save(subcourse);
```

Comment:
what has changed in the subcourse that we need to save it?

Reply:
We added the lecture to the subcourse and save that change. This could be dropped in favour of specifying the lectures subcourse on construction. But then the lecture's constructor would be somewhat independent of the subcourse model and could be located elsewhere. In the spirit of "adding a lecture to a subcourse" I decided to append the lectures array in the subcourse by the new lecture. But this is certainly up to taste 🙂

Reply:
Sorry for the delay! This looks good to me as well!